https://bugzilla.redhat.com/bugzilla/show_bug.cgi?id=190810


2006-09-26  Jan Kratochvil  <jan.kratochvil@redhat.com>

	* remote.c (remote_wait): Suggestion on crash due to nonmatching target.
	(remote_async_wait): Likewise.


Index: gdb-6.8/gdb/remote.c
===================================================================
--- gdb-6.8.orig/gdb/remote.c	2008-07-14 10:27:07.000000000 +0200
+++ gdb-6.8/gdb/remote.c	2008-07-14 10:27:17.000000000 +0200
@@ -3509,8 +3509,13 @@ Packet: '%s'\n"),
 					 reg->regnum, regs);
 		  }
 
+		/* It may also occur on amd64 which defaults to 32-bit i386
+		   target.  gdbserver(1) is not aware of the `set architecture'
+		   name itself as it is not using libbfd.  */
 		if (*p++ != ';')
-		  error (_("Remote register badly formatted: %s\nhere: %s"),
+		  error (_("Remote register badly formatted: %s\nhere: %s"
+			   "\nTry to load the executable by `file' first,"
+			   "\nyou may also check `set/show architecture'."),
 			 buf, p);
 	      }
 	  }
@@ -3737,8 +3742,13 @@ Packet: '%s'\n"),
 					 reg->regnum, regs);
 		  }
 
+		/* It may also occur on amd64 which defaults to 32-bit i386
+		   target.  gdbserver(1) is not aware of the `set architecture'
+		   name itself as it is not using libbfd.  */
 		if (*p++ != ';')
-		  error (_("Remote register badly formatted: %s\nhere: %s"),
+		  error (_("Remote register badly formatted: %s\nhere: %s"
+			   "\nTry to load the executable by `file' first,"
+			   "\nyou may also check `set/show architecture'."),
 			 buf, p);
 	      }
 	  }
